ghidrall/src/lifter.py

```diff
@@ -54,9 +55,9 @@ def create_function_signatures(self):
             xml_return_type = function_xml.find('return')
             return_types[function] = "void"
             if xml_return_type.find('type').text == "undefined8":
-                func_return = ir.VoidType()
+                func_return = void_type
```

Comment:
this seems wrong. `undefined8` is some `8` byte quantity, like `uint64_t`. Not same as `void`

Reply:
Hmm. That's a good point. I think the issue here is that there isn't really a one to one mapping between llvm data types and ghidra data types so we kind of have to choose the closest. I think as of now, VoidType is kind of the default data type when we aren't really sure about the type, as in the case of undefined8. So although using VoidType causes us to lose some information with regards to the size, it takes care of the fact that we don't really know what type is best. As of now, I think VoidType is the best of the data types available in llvm. What do you think?

Reply:
Ghidra attempts to return 0 instead of returning void in many cases, for whatever reason. Unspecified8 is the type that holds 0 when it performs that copy and return.

Reply:
`void` is not a type, it is an absence of type. There is an immediate map from ghidra integral types to LLVM integral types. In particular, `unspecified8` should map to `i64` in LLVM, where `i64` is integral type of 64bits. LLVM names types by bits, but ghidra by bytes, but that is the only difference. There are really only three types, void, integer, i.e., iXXX, and pointer, i.e., `i8*`